src/test-suites/null-keys/b.subgraph.ts

```diff
@@ -0,0 +1,44 @@
+import { createSubgraph } from "../../subgraph.js";
+import { books } from "./data.js";
+
+export default createSubgraph("b", {
+    typeDefs: /* GraphQL */ `
+    extend schema
+      @link(
+        url: "https://specs.apollo.dev/federation/v2.3"
+        import: ["@key"]
+      )
+
+    type Book @key(fields: "id") @key(fields: "upc") {
+        id: ID!
+        upc: ID!
+    }
+  `,
+    resolvers: {
+        Book: {
+            __resolveReference(reference: { id: String; } | { upc: String; }) {
+                if (reference != null) {
+                    let book: { id: string; upc: string; } | undefined;
+                    if ('id' in reference) {
+                        book = books.find((book) => book.id === reference.id);
+                    }
+                    if ('upc' in reference) {
+                        book = books.find((book) => book.upc === reference.upc);
+                    }
+                    if (book != null) {
+                        // Return this null on purpose
```

Comment:
on what purpose? What's the purpose?

Reply:
Can we explain here why it returns `null` for a Book with upc === 'b3'?

The query starts from `Query.bookContainers`. The `id` needs to be resolved from `upc`, in order to resolve Book.author. Because Book with upc === 'b3' is not resolved, the author is null.

Reply:
The scenario is;
`a` has `Book` entities with `upc`s `b1`, `b2` and `b3`
But `b` has `Book` entities with only `b1` and `b2`